src/sonic-device-data/tests/media_checker

```diff
@@ -46,17 +49,32 @@ def check_media_dict(vendor_dict):
             print("Expecting settings for vendor type " + vendor_key)
             return False
 
+        lane_speed_hierarchy_exists = None
+
         for value_key in value_dict:
-            if value_key not in setting_keys:
-                print("Unknown media setting " + value_key)
-                return False
+            if value_key.startswith(lane_speed_key_prefix):
+                if lane_speed_hierarchy_exists == False:
+                    print("Inconsistent lane speed hierarchy levels for the same entry")
+                    return False
+                lane_speed_hierarchy_exists = True
+                settings_dict = value_dict[value_key]
+            else:
+                if lane_speed_hierarchy_exists == True:
+                    print("Inconsistent lane speed hierarchy levels for the same entry")
+                    return False
+                lane_speed_hierarchy_exists = False
+                settings_dict = value_dict
 
-            lane_dict = value_dict[value_key]
-            for lanes in lane_dict:
-                if not check_lane_and_value(lanes, lane_dict[lanes]):
+            for settings_key in settings_dict:
```

Comment:
```suggestion
            for si_param in si_params_list:
```

Reply:
'si_params_list' and 'settings_dict' are not interchangeable because they store different data: 'si_params_list' only contains the names of the SI parameters, while 'settings_dict' maps each SI parameter name to its corresponding list of SI values.
Changing the loop variable name to 'si_param' is a good idea, so I'm renaming it.